app/assets/v2/js/cart.js

```diff
@@ -1003,148 +1000,204 @@ Vue.component('grants-cart', {
+          const tokenAddress = isEth ? '0x0000000000000000000000000000000000000000' : tokenDetails.addr;
+
+          // Replace undefined comments with empty strings
+          const comment = donation.grant.grant_comments === undefined ? '' : donation.grant.grant_comments;
+
+          // For automatic contributions to Gitcoin, set 'gitcoin-grant-input-amount' to 100.
+          // Why 100? Because likely no one will ever use 100% or a normal grant, so using
+          // 100 makes it easier to search the DB to find which Gitcoin donations were automatic
+          const isAutomatic = donation.grant.isAutomatic;
+          const gitcoinGrantInputAmt = isAutomatic ? 100 : Number(this.gitcoinFactorRaw);
+
+          // Add the donation parameters
+          saveSubscriptionPayload.admin_address.push(donation.grant.grant_admin_address);
+          saveSubscriptionPayload.amount_per_period.push(Number(donation.grant.grant_donation_amount));
+          saveSubscriptionPayload.comment.push(comment);
+          saveSubscriptionPayload.confirmed.push(false);
+          saveSubscriptionPayload.contract_address.push(donation.grant.grant_contract_address);
+          saveSubscriptionPayload.contract_version.push(donation.grant.grant_contract_version);
+          saveSubscriptionPayload.denomination.push(tokenAddress);
+          saveSubscriptionPayload['gitcoin-grant-input-amount'].push(gitcoinGrantInputAmt);
+          saveSubscriptionPayload.grant_id.push(grantId);
+          saveSubscriptionPayload.split_tx_id.push(txHashes[i]);
+          saveSubscriptionPayload.sub_new_approve_tx_id.push(donation.tokenApprovalTxHash);
+          saveSubscriptionPayload.token_address.push(tokenAddress);
+          saveSubscriptionPayload.token_symbol.push(tokenName);
+        } // end for each donation
+
+        // to allow , within comments
+        saveSubscriptionPayload.comment = saveSubscriptionPayload.comment.join('_,_');
+
+        // Configure request parameters
+        const url = '/grants/bulk-fund';
+        const headers = {
+          'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8'
+        };
+        const saveSubscriptionParams = {
+          method: 'POST',
+          headers,
+          body: new URLSearchParams(saveSubscriptionPayload)
+        };
 
-      for (let i = 0; i < donations.length; i += 1) {
-        // Get URL to POST to
-        const donation = donations[i];
-        const grantId = donation.grant.grant_id;
+        // Send saveSubscription request
+        const res = await fetch(url, saveSubscriptionParams);
+        const json = await res.json();
+
+        MauticEvent.createEvent({
+          'alias': 'products',
+          'data': [
+            {
+              'name': 'product',
+              'attributes': {
+                'product': 'grants',
+                'persona': 'grants-contributor',
+                'action': 'contribute'
+              }
+            }
+          ]
+        });
```

Comment:
Is this an analytics thing? It occurs after the `fetch` so I don't think it should be an issue, but just wanted to confirm there's no way this could cause issues similar to how saving JSON stores caused issues

Reply:
Yup this is an analytics thing ! @octavioamu would it be worth doing this after the POST call is complete
@mds1 has a valid point here

Reply:
yes is a post to our internal endpoint and from there to mautic.